### PR TITLE
many: name unversioned data.

### PR DESCRIPTION
**Pull request by kyrofa**
Currently the unversioned data directory is available for use, but not represented by an environment variable for easy use. In order to use it, one must use $SNAP_DATA/../common or $SNAP_USER_DATA/../common. This PR fixes LP: [#1577549](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1577549) by naming them $SNAP_COMMON and $SNAP_USER_COMMON. It also adds Spread tests to verify that all the data directories are created and can be written to by apps and services.

### CODE REVIEW

File: tests/writable-areas/task.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  rm data-writer_1.0_all.snap
+execute: |
+  snap install data-writer_1.0_all.snap
+
+  echo "Apps can write to writable areas"
+  data-writer.app
+  [ -f /var/snap/data-writer/x1/from-app ] || exit 1
+  [ -f /var/snap/data-writer/common/from-app ] || exit 1
+  [ -f /root/snap/data-writer/x1/from-app ] || exit 1
+  # TODO: As soon as `snap run` is used (which creates this directory),
```
Comment (mvo5) [inline tests/writable-areas/task.yaml]: You could just call `snap run data-writer.app` here yourself :)

**Comment by mvo5** [comment]
Looks good, one suggestion for the spread test

File: tests/writable-areas/task.yaml (hunk 1 above)
Comment (kyrofa) [inline tests/writable-areas/task.yaml]: Yeah but then it's not quite testing as much as it should. And the service will still be missing it, so might as well keep them consistent. What do you think?

Comment (mvo5) [inline tests/writable-areas/task.yaml]: Yeah, its fine.

**Comment by niemeyer** [comment]
Breakage seems real, but unrelated, :
```
error: cannot perform the following tasks:
- Download snap "hello-world" from channel "stable" (snap not found)
```
```
2016/06/23 21:03:43 Failed tasks: 2
    - linode:ubuntu-16.04-32-grub:tests/main/install-store
    - linode:ubuntu-16.04-64-grub:tests/main/install-store
```

**Pull request merged**